### ISSUE TITLE
Incomming message with more than 13kb breaks signalr hub.

### ISSUE THREAD

**Issue by bleekbless** [aspnet/SignalR#2266]
### Please include as much of the following as you can in your bug report

* Versions of Server-Side NuGet Packages: Microsoft.AspNetCore.SignalR 1.0.0-rc1-final 
* Versions of Client-Side NuGet/NPM Packages: @aspnet/signalr 1.0.0-rc1-update1
* Are you using the C# client or the JavaScript client: JavaScript Client
* The Server you are using (Kestrel/HttpSysServer/IIS/IIS Express/Azure Web App/etc.): Kestrel
* The Operating System on the Server (Windows/Linux/macOS): Linux
* The Operating System on the Client (Windows/Linux/macOS): Linux
* The Browser on the client, if using the JavaScript client (IE/Chrome/Edge/Firefox/etc.): Chrome
* DEBUG BACKEND --> [debug.txt](https://github.com/aspnet/SignalR/files/2001030/debug.txt)

What happens is when I send a message with a base64 file attached the HUB breaks and don't proccess the message.

When in doubt, feel free to file the issue, we're happy to help answer questions. We also suggest using the `asp.net-core-signalr` tag on StackOverflow to ask questions.

**Comment by davidfowl** [comment]
Yes, we need to think about exposing a first class message limit to SignalR itself so that the error message doesn't suck (right now it's coming from the transport layer). For sending larger message payloads, you'd want streaming support from client to server which we don't have yet but is planned for a future release. Today SignalR needs to buffer the entire message before it calls your hub and it doesn't start consuming the buffer until it seems an entire message.

> What happens is when I send a message with a base64 file attached the HUB breaks and don't proccess the message.

SignalR isn't ideal for sending files until we get streaming support. 

That said, you can bump the maximum size by setting https://github.com/aspnet/SignalR/blob/36b7f72460cfdd49e6b2c5d97e03fff719633ff8/src/Microsoft.AspNetCore.Http.Connections/HttpConnectionDispatcherOptions.cs#L59.

```C#
app.UseSignalR(routes =>
{
    routes.MapHub<ChatHub>("/chat", options =>
    {
        // 30Kb message buffer
        options.ApplicationMaxBufferSize = 30 * 1024 * 1024;
    });
});
```

The default is 32 KiB

https://github.com/aspnet/SignalR/blob/36b7f72460cfdd49e6b2c5d97e03fff719633ff8/src/Microsoft.AspNetCore.Http.Connections/HttpConnectionDispatcherOptions.cs#L16

Bare in mind this accounts for the entire payload, not just your argument data. Also, beware of setting the limits too high as it could lead to memory bloat.

**Comment by e-davidson** [comment]
I had the same issue.

Thank you @davidfowl  your solution worked.

**Comment by popcatalin81** [comment]
Used the following configuration to get rid of the error:

```csharp
routes.MapHub<AgentHub>("/ws/Agent", options => 
{
     options.ApplicationMaxBufferSize = 0;
     options.TransportMaxBufferSize = 0;
});
```

I know this looks like someon shooting itself it the foot. But it's not. It's exactly the configuration needed for this particualr project where 99.999% of the messages are small and only 0.0001% are large and where a performace degradation is the correct tradeoff instead of an error. (Geting an error is not the right choise here, is catastrophic actually)

**Comment by davidfowl** [comment]
@popcatalin81 I'd also leave the TransportMaxBufferSize as it shouldn't have the same problem (we never partially consume the output buffer).

**Comment by popcatalin81** [comment]
> You might be introducing a security hole into your system if you allow arbitrary clients to connect from the internet

We actually "are sending files with Regular HTTP post requests" not SignalR. The connections are secured with both Server and Client authentication.. 

This is a private API using .Net Core Client SignalR libraries and AspNet Core SignalR server.  The environment is pretty well controlled, except for some very rare large packets (that we could split, but the effect would be to combine them back in memory, same as using unbounded buffers).

**Comment by popcatalin81** [comment]
Unfortunately I can't give an example here as I'm under the effects of an NDA. I can give you an example it remains confidential and you give me an email address to send it to.

**Comment by davidfowl** [comment]
Oh snap thanks.
